src/searchReplace.tsx

```diff
@@ -52,10 +58,56 @@ export class SearchReplaceModel extends VDomModel {
   }
 
   private _searchString: string;
-  private _queryResults: any;
+  private _queryResults: IResults[];
   private _debouncedStartSearch: Debouncer;
 }
 
+interface IQueryResult {
+  matches: IResults[];
+}
+
+interface IResults {
+  path: string;
+  matches: {
+    line: string;
+    start: number;
+    end: number;
+    match: string;
+    line_number: number;
+    absolute_offset: number;
+  }[];
+}
+
+function createTreeView(results: IResults[]): JSX.Element {
+  const items = results.map(file => {
```

Comment:
We need to sort the results in a deterministic way. A good sorting will be to sort per path level and within a directory per filename.